### ISSUE TITLE
Import from blissdata instead of bliss

### ISSUE THREAD

**Issue by woutdenolf** [vasole/pymca#905]
The [`blissdata`](https://gitlab.esrf.fr/bliss/blissdata) packages is a much lighter dependency than `bliss`. It is available on pypi and anaconda.

```python
from bliss.config import get_sessions_list
from bliss.config.settings import scan as rdsscan
from bliss.data.node import get_node, get_nodes
```

```python
from blissdata.settings import scan as rdsscan
from blissdata.data.node import get_node, get_nodes
```

`get_sessions_list` is unfortunately not in `blissdata`. I'll try to make that happen.

**Comment by vasole** [comment]
Did you get a get_sessions_list replacement? It is to merge #1039 and get rid of the deprecation warning.

**Comment by woutdenolf** [comment]
> personally I would make it as a command line option, to specify the list of sessions we are
interested in. No discovery.

Discovery is mandatory imo.

**Comment by mguijarr** [comment]
> > personally I would make it as a command line option, to specify the list of sessions we are
> > interested in. No discovery.
> 
> Discovery is mandatory imo.

I see it as a problem, for people who do not use Beacon or BLISS yaml configuration files. How to identify a session?

It could also be, that the "Data source" button prompts user for the session name.  Users know the name of the
BLISS session they use. The successful attempts to connect to a session could then be stored in PyMCA local
preferences configuration (if it exists ?), to provide selection via a combo box to speed up next times...

It would be nice to take into account those who do not have Beacon nor BLISS yaml configuration files. So, if BEACON_HOST is undefined it would look at REDIS_DATA_URL to know how to
reach Redis.

**Comment by woutdenolf** [comment]
Sorry, I don't agree with pushing complexity onto third-party applications. Blissdata could easily provide a function that returns a list of currently available BLISS sessions.

**Comment by mguijarr** [comment]
Ok.

What do you propose in the case when there is no Beacon ? No YAML files, no way to know sessions. Maybe it is possible to discover sessions from recorded data in Redis ?

**Comment by vasole** [comment]
I agree with @woutdenolf that the available sessions should be discoverable.

BTW, just by chance does `pymca -f --spec=NAME_OF_BLISS_SESSION --shm=None` help?

**Comment by mguijarr** [comment]
Well, at the moment newest blissdata >= 1.0.0 does not work at all because those imports fail:

```
from blissdata.settings import scan as rdsscan
from blissdata.data.node import get_node, get_nodes
```

(in src/PyMca5/PyMcaCore/RedisTools.py )

So, I suppose it does not work at ESRF neither - people use files apparently.

**Comment by mguijarr** [comment]
> available sessions should be discoverable.

I had a deeper look at this - blissdata can already communicate with Beacon (the beamline configuration server),
so far so good, so we make queries. As of today, though, there is no knowledge of BLISS objects (`Session` is one) on the server side. Everything is interpreted in the client (which is good, and bad, depending what has to be done !).

I think blissdata should not import bliss itself. So, the client logic has to be reimplemented in blissdata for finding sessions from yaml files contents.

But it is not trivial. `get_sessions_list` code does:

```
    for name in config.names_list:
        c = config.get_config(name)
        if c.get("class") != "Session":
            continue
        if c.get_inherited("plugin") != "session":
            continue
        all_sessions.append(name)
```

It is going through indexed elements (pieces of yaml with `name: something`), checking if a `class` member with `Session` is present ; it also checks if it is the good configuration loader (aka "plugin"), by looking in the inherited
fields of the indexed element... Maybe this last check could be skipped to simplify.

A better bet is maybe to rely on the Configuration web application. It is always present at ESRF beamlines.
In this case getting sessions list is as easy as a GET request: http://localhost:9030/config/tree/sessions

Personally I would go for this for having sessions discoverable from blissdata.

Still what to do if there is no Beacon ?
I see PyMca already has the third-party application logic pushed to it with `--spec` and `--no-shm` so if it is not desirable to add `--bliss` , we could do as suggested by @vasole and hijack `--spec` for BLISS sessions explicitely specified on the command line.

**Comment by vasole** [comment]
Well, ESRF people were never encouraged to use PyMca for online data despite offering features not offered by flint. So I do not find surprising that something got broken with recent bliss developments.

The funny thing is that today PyMca can access bluesky data but not bliss.

**Comment by mguijarr** [comment]
> The funny thing is that today PyMca can access bluesky data but not bliss.

Are you referring to https://github.com/vasole/pymca/pull/1064 ?

It is not merged, and only a draft.

I would like to fix the direct BLISS support, the code in RedisTools.py is less than 400 lines... I am not very familiar with this, though.

**Comment by mguijarr** [comment]
What is the link between PyMca and silx ? I mean, if it brings more benefits (like, having it compatible with `silx view` for example) instead of having BLISS support in PyMca like it was done, it would maybe be useful to go through silx like it is done for Tiled ?

I am a bit lost.

**Comment by vasole** [comment]
Hi,

The direct reading via blissdata should be restored.

I guess it got broken when trying to use blissdata:

https://github.com/vasole/pymca/pull/1039

Perhaps using the direct access to Redis still works.

**Comment by mguijarr** [comment]
> Perhaps using the direct access to Redis still works.

No, it's not possible it has completely changed - different way to organize data, different structures, different keys... Everything

**Comment by vasole** [comment]
OK. Then the access via blissdata has to work,.

**Comment by vasole** [comment]
The implemented support was very basic. It was dealing with the equivalent of the SPEC file part of the data (motors, counters and MCAs) because I was wrapping as a SPEC file. If you know how to obtain that information (scans, motors, counters and MCAs) via blissdata it should be easy to adapt RedisTools.py

**Comment by woutdenolf** [comment]
> What do you propose in the case when there is no Beacon ?

Why Beacon? Everything is in Redis.
